attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/service/AttributeMaskingServiceErrorTest.java

```diff
@@ -0,0 +1,206 @@
+import uk.gov.gchq.palisade.service.attributemask.model.AuditableAttributeMaskingRequest;
+import uk.gov.gchq.palisade.service.attributemask.model.AuditableAttributeMaskingResponse;
+import uk.gov.gchq.palisade.service.attributemask.repository.PersistenceLayer;
+import uk.gov.gchq.palisade.service.attributemask.service.AttributeMaskingAspect;
+import uk.gov.gchq.palisade.service.attributemask.service.AttributeMaskingService;
+import uk.gov.gchq.palisade.service.attributemask.service.LeafResourceMasker;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(classes = AttributeMaskingApplication.class, webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@ContextConfiguration(classes = {ExecutorTestConfiguration.class, AttributeMaskingServiceErrorTest.Config.class})
+@EntityScan(basePackageClasses = {AuthorisedRequestEntity.class})
+@EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.attributemask.repository"})
+class AttributeMaskingServiceErrorTest {
+
+    private static final Function<Integer, String> REQUEST_FACTORY_JSON = i -> String.format("{\"userId\":\"test-user-id\",\"resourceId\":\"/test/resourceId%d\",\"context\":{\"class\":\"uk.gov.gchq.palisade.Context\",\"contents\":{\"purpose\":\"test-purpose\"}},\"user\":{\"userId\":{\"id\":\"test-user-id\"},\"roles\":[],\"auths\":[],\"class\":\"uk.gov.gchq.palisade.User\"},\"resource\":{\"class\":\"uk.gov.gchq.palisade.resource.impl.FileResource\",\"id\":\"/test/resourceId\",\"attributes\":{},\"connectionDetail\":{\"class\":\"uk.gov.gchq.palisade.service.SimpleConnectionDetail\",\"serviceName\":\"test-data-service\"},\"parent\":{\"class\":\"uk.gov.gchq.palisade.resource.impl.SystemResource\",\"id\":\"/test/\"},\"serialisedFormat\":\"avro\",\"type\":\"%d\"},\"rules\":{\"message\":\"no rules set\",\"rules\":{\"test-rule\":{\"class\":\"uk.gov.gchq.palisade.contract.attributemask.ContractTestData$PassThroughRule\"}}}}", i, i);
+
+    private final Function<Integer, JsonNode> requestFactoryNode = i -> {
+        try {
+            return this.mapper.readTree(REQUEST_FACTORY_JSON.apply(i));
+        } catch (JsonProcessingException e) {
+            throw new SerializationFailedException("Failed to parse contract test data", e);
+        }
+    };
+
+    private final Function<Integer, AttributeMaskingRequest> requestFactoryObj = i -> {
+        try {
+            return this.mapper.treeToValue(requestFactoryNode.apply(i), AttributeMaskingRequest.class);
+        } catch (JsonProcessingException e) {
+            throw new SerializationFailedException("Failed to convert contract test data to objects", e);
+        }
+    };
+
+    @Autowired
+    private AttributeMaskingService attributeMaskingService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    void persistenceFailureTest() {
```

Comment:
test names should be formatted to "testPersistenceFailure", sonarQube should be throwing a code smell here